### PR TITLE
Better organizing of utils functions

### PR DESCRIPTION
**Pull request by ngreenwald**
## Major change 1
Reorganized the functions in reshape_data.py into more logical utils scripts where appropriate. 

The workflow for our data reshaping is as follows:
1. We take paired image and label data
2. We decide if it needs to be chopped into smaller rows/cols crops
3. We decide if it needs to be chopper into smaller z-stack slices
4. We chop and slice the data accordingly. 
5. We save the data to NPZs for caliban. 

It then gets labeled/corrected. Once we have the data back, the workflow is as follows:
1. We load the data from NPZs
2. If the data was sliced into smaller z-stacks, we stitch them back together
3. If the data was cropped into smaller rows/cols pieces, we stitch them back together

This used to all be in the reshape_data.py script. I have now reorganized it as follows. 

- All the functions relating to cropping are in utils/crop_utils
- All the functions relating to slicing z-stacks are in utils/slice_utils
- All of the functions for saving and loading the NPZs are in utils/io_utils
- The high-level functions for reshaping the data are still in reshape_data

This entire pipeline interacts with data that is of the following format:
[fovs, stacks, crops, slices, rows, cols, channels]. 

- The fov index (index 0) is used to access each unique image FOV. So if there were multiple experiments in different physical locations, these can be stored in a single array as different FOVs
- The stack index (index 1) is used to access each unique image slice. So if you have a z-stack with 30 slices over time, each entry in the stack index represents one of those slices
- The crops index (index 2) is used to access a unique crop of an image. So if you've reshaped a dataset that used to be rows x cols of 1024x1024, and is now 256x256, crops will be an index of length 16 containing an index for each crop. Uncropped data will have a length of 1 in the cropped dimension
- The slices index (index 3) is used to access a unique slice of a z-stack. So if you had a z-stack of length 30, and you chopped it into 3 different stacks of length 10, slices will be of length 3, to access each of those distinct image stacks
- The rows index (index 4) is the rows of the image data
- The cols index (index 5) is the cols of the image data
- The channels index (index 6) is the channels of the data

This format is convenient because no matter what stage of the cropping, slicing, stitching functions we're in, the data is always of the same shape with the same dimensions, the values within each dimension just get modified as appropriate. 

## Major change 2
The pipeline previously used a single array of length channels + 1 to store both the image data and the labeled data. This was hacky, and was leading to issues. Especially as we expand to multiple channels. I modified all of the functions to instead expect image and label data (X and y) to be present in separate arrays, in line with how they are treated in deepcell. 

## Minor changes
I brought in pad_xr_dims and other helper functions to utils/data_utils, which previously were in the angelolab repo
I modified some of the test functions to be more clear

### CODE REVIEW

**Comment by willgraf** [comment]
Can you go into more detail in the PR body? With 3k+ lines of changed code that I'm not familiar with, I'm having trouble giving this a good review. (Also why I added @MekWarrior as a reviewer).

**Comment by ngreenwald** [comment]
Definitely, sorry about that. See revised explanation

File: caliban_toolbox/aws_functions.py [hunk 1]
```diff
@@ -91,7 +91,9 @@ def aws_upload_files(aws_folder, stage, upload_folder, pixel_only, label_only, r
     subfolders = re.split('/', aws_folder)
     subfolders = '__'.join(subfolders)
 
-    url_dict = {'pixel_only': pixel_only, 'label_only': label_only, 'rgb': rgb_mode}
+    # TODO: add back label_only once logic is working correctly in caliban
```
Comment (MekWarrior) [inline caliban_toolbox/aws_functions.py:94]: Shouldn't this be added back now as the label_only issue was closed by Caliban #163?

File: caliban_toolbox/utils/plot_utils.py [hunk 2]
```diff
@@ -132,51 +134,46 @@ def overlay_crop_overlap(img_crop, row_starts, row_ends, col_starts, col_ends):
...
 
-def set_channel_colors(combined_xr, plot_colors):
+def set_channel_colors(channel_data, plot_colors):
     """Modifies the order of image channels so they're displayed with appropriate color in caliban
 
     Args:
-        combined_xr: xarray containing channels and labels
-        plot_colors: array containing the color of each channel, in order of the current channels
+        channel_data: xarray containing channels
+        plot_colors: array containing the desired color for each channel
```
Comment (MekWarrior) [inline caliban_toolbox/utils/plot_utils.py:142]: Where do we handle the colors for the labels now that they are split out from the xarray data?

Comment (ngreenwald) [reply caliban_toolbox/utils/plot_utils.py:142]: Caliban has a color map that determines how label colors are handled. The channel colors are determined by their index in the RGBCMY array that is used for plotting. So the above function puts the channel in right order for the desired color, but leaves it up to caliban to set the color for a specific label.

File: caliban_toolbox/aws_functions.py (hunk 1 above)
Comment (ngreenwald) [reply caliban_toolbox/aws_functions.py]: Yup! Done

**Comment by MekWarrior** [comment]
the inits still need to be fixed, but as this PR moves around a number of util functions and the data loader may add more, we will add that as a TODO item for the next PR.

**Pull request merged**